src/InputOutput/readers.jl

```diff
@@ -535,14 +585,16 @@ function read_field(reader::HDF5Reader, name::AbstractString)
         # For when `Nh` is added back to the type space
         #     Nhd = Nh_dim(data_layout)
         #     Nht = Nhd == -1 ? () : (size(data, Nhd),)
-        ElType = eval(Meta.parse(attrs(obj)["value_type"]))
+        ElType = read_type(attrs(obj)["value_type"])
```

Comment:
```suggestion
        ElType = read_type(attrs(obj)["field_eltype_type"])
```
and can we move this to right after `elseif data_layout in ("DataF",)`?

Reply:
This change makes this PR a breaking change. Old hdf5 files containing `fields` will no longer be able to be read unless the  "value_type" attribute is changed to "field_eltype".

Reply:
Ah, I see. I guess it's fine to leave it for now. We could rename it and preserve backward compatibility, but I don't want to hold up this work.